core/src/com/pottda/game/controller/ControllerHookup.java

```diff
@@ -53,12 +53,14 @@ public void onNewModel(ModelActor m) {
             view = new ActorView(m.sprite, ((Vector2f) ((Obstacle) m).size));
             controller = new ObstacleController(m, view);
         }
-        if(view != null) {
+        
+        try {
             stage.addActor(view);
-        }
-        if (controller != null) {
             notifyListeners(controller);
         }
+        catch (NullPointerException e){
+            throw new Error("couldn't handle model-type",e);
```

Comment:
the view not being created is because we can't handle the model type, i.e. its the same cause/outcome